File: 1-100/21-30/26-removeDuplicate/removeDuplicate.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        index = len(nums) - 1
        chr = '.' # start point
        
        while index >= 0:
            if chr != nums[index]:
                chr = nums[index]
            else:
                del nums[index]
            index -= 1
            
        return len(nums)
            

File: 1-100/21-30/26-removeDuplicate/removeDuplicate.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int storeIndex = 0;

        for (auto &it: nums)
            if (storeIndex == 0 || it != nums[storeIndex-1])
                nums[storeIndex++] = it;

        return storeIndex;
    }
};